From 496612c047f05ee86423d0db75f54a9173832905 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Mon, 13 Oct 2014 22:30:01 -0400
Subject: [misc] genwqe: fix return value check in genwqe_device_create()

Message-id: <1413239416-7778-14-git-send-email-gduarte@redhat.com>
Patchwork-id: 97544
O-Subject: [RHEL7.1 PATCH BZ 1088491 13/28 v2] misc: genwqe: fix return value check in genwqe_device_create()
Bugzilla: 1088491
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088491
Upstream Status: 634608f27acd098b245ca6fe60e06701185eb170

commit 634608f27acd098b245ca6fe60e06701185eb170
Author: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date:   Tue Jan 7 16:56:35 2014 +0800

    misc: genwqe: fix return value check in genwqe_device_create()

    In case of error, the function device_create_with_groups()
    returns ERR_PTR() and never returns NULL. The NULL test in
    the return value check should be replaced with IS_ERR().

    Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
    Acked-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/genwqe/card_dev.c b/drivers/misc/genwqe/card_dev.c
index 91a1c15..8f8a6b3 100644
--- a/drivers/misc/genwqe/card_dev.c
+++ b/drivers/misc/genwqe/card_dev.c
@@ -1306,8 +1306,8 @@ int genwqe_device_create(struct genwqe_dev *cd)
          genwqe_attribute_groups,
          GENWQE_DEVNAME "%u_card",
          cd->card_idx);
- if (cd->dev == NULL) {
-  rc = -ENODEV;
+ if (IS_ERR(cd->dev)) {
+  rc = PTR_ERR(cd->dev);
   goto err_cdev;
  }
 
-- 
1.7.1